snippets/csharp/VS_Snippets_Winforms/ContainerExmpl/CS/librarycontainer.cs

```diff
@@ -228,22 +228,15 @@ static void Main(string[] args)
 		{
 			LibraryContainer cntrExmpl = new LibraryContainer();
 
-			try
-			{
-				BookComponent book1 = new BookComponent("Wizard's First Rule", "Terry Gooodkind");
-				cntrExmpl.Add(book1, "0812548051");
-				BookComponent book2 = new BookComponent("Stone of Tears", "Terry Gooodkind");
-				cntrExmpl.Add(book2, "0812548094");
-				BookComponent book3 = new BookComponent("Blood of the Fold", "Terry Gooodkind");
-				cntrExmpl.Add(book3, "0812551478");
-				BookComponent book4 = new BookComponent("The Soul of the Fire", "Terry Gooodkind");
-				//This will generate exception because the ISBN already exists in the container.
-				cntrExmpl.Add(book4, "0812551478");
-			}
-			catch(SystemException e)
-			{
-				Console.WriteLine("Error description: " + e.Message);
-			}
+			BookComponent book1 = new BookComponent("Wizard's First Rule", "Terry Gooodkind");
```

Comment:
Since the container doesn't support duplicates, we can't simply not handle exceptions. The problem here is that line 184 throws a SystemException, which is inappropriate. You could change it to ArgumentException, then handle the ArgumentException in a try/catch block around all of the BookComponent instantiates and calls to the Add method.